src/Parser.js

```diff
@@ -834,7 +836,7 @@ export default class Parser {
         if (this.nextToken.text === (optional ? "[" : "{")) {
             // If we get a brace, parse an expression
             this.consume();
-            const expression = this.parseExpression(false, optional ? "]" : null);
+            const expression = this.parseExpression(false, optional ? "]" : "}");
```

Comment:
I'm not sure why this was `null`.

Reply:
Parsing always ends on `}`, so I'm not sure specifying `"}"` is necessary... hence the `null`.  (It's been a while since I've looked at this code, so I forget the exact details...)

Reply:
It feels nice to have symmetry the `? "[" : "{"` check above.